Implementation/src/main/java/xmlparser/XMLparser.java

```diff
@@ -0,0 +1,162 @@
+        createDoc(path);
+
+        if(doc == null){
+            System.out.println("Document wasn't created");
+            return;
+        }
+        NodeList mdbNodeList = doc.getElementsByTagName(MITGLIED_DEUTSCHER_BUNDESTAG_TAG);
+
+        List<Person> personen = new ArrayList<>();
+        List<Redner> rednerList = new ArrayList<>();
+
+        for (int i = 0; i < mdbNodeList.getLength(); i++) {
+            Node mdbNode = mdbNodeList.item(i);
+            if(mdbNode.getNodeType() == Node.ELEMENT_NODE){
+                Element mdbElement = (Element) mdbNode;
+                int id = Integer.parseInt(getPersonID(mdbElement));
+                Map<String,String> personProperties = getNameProperties(mdbElement);
+                Map<String, String> bioDataProperties = getBioData(mdbElement);
+                Person person = new Person(personProperties.get(AKAD_TITEL_TAG), personProperties.get(VORNAME_TAG), personProperties.get(NACHNAME_TAG),
+                        bioDataProperties.get(BERUF_TAG), bioDataProperties.get(GESCHLECHT_TAG), bioDataProperties.get(GEBURTSDATUM_TAG), bioDataProperties.get(FAMILIENSTAND_TAG), bioDataProperties.get(RELIGIO_TAG), bioDataProperties.get(GEBURTSORT_TAG), null);
+                personen.add(person);
+                rednerList.add(new Redner(id, person));
+            }
+
+        }
+        System.out.println("");
+    }
+
+    public void parseProtocol(String path){
+        createDoc(path);
+
+        if(doc == null){
+            System.out.println("Document wasn't created");
+            return;
+        }
+
+        Element dbtplenarprotokoll = doc.getDocumentElement();
+        NodeList rednerList = doc.getElementsByTagName("rednerliste");
+
+        for (int i = 0; i < rednerList.getLength(); i++){
+            Node rednerListNode = rednerList.item(i);
+            if(rednerListNode.getNodeType() == Node.ELEMENT_NODE){
+                Element eElement = (Element) rednerListNode;
+                NodeList redner = eElement.getElementsByTagName("redner");
+                for (int j = 0; j < redner.getLength(); j++) {
+                    Node rednerNode = redner.item(j);
+                    String rednerID = ((Element)rednerNode).getAttribute("id");
+
+                }
+            }
+        }
+    }
+
+    private void createDoc(String path){
+        if(path == null || path.equals("")){
+            System.out.println("No file name was given");
+            return;
+        }
+        try{
+            file = new File(new File("").getAbsolutePath() + path);
+            System.out.println(file.getAbsolutePath());
+
+            if(!file.exists()){
+                System.out.println("File doesn't exist");
+                return;
+            }
+
+            DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
+            DocumentBuilder dbBuilder = dbFactory.newDocumentBuilder();
+            doc = dbBuilder.parse(file);
```

Comment:
Have you tested this generic Parse() function on different files (from the 18. Legislaturperiode)? What about the BaseData? Does it work properly on huge files? What are the limits?